src/pages/DistributionsPage/useLockedTokenDistributions.tsx

```diff
@@ -0,0 +1,103 @@
+import assert from 'assert';
+
+import { parseUnits } from 'ethers/lib/utils';
+import { lockedTokenDistribution } from 'lib/hedgey';
+
+import { DebugLogger } from '../../common-lib/log';
+import { useContracts } from 'hooks';
+
+import {
+  useMarkLockedDistributionDone,
+  useSaveLockedTokenDistribution,
+} from './mutations';
+import { getProfileIds } from './queries';
+
+const logger = new DebugLogger('useLockedTokenDistribution');
+
+export const useLockedTokenDistribution = () => {
+  const contracts = useContracts();
+  const { mutateAsync: saveLockedTokenDistribution } =
+    useSaveLockedTokenDistribution();
+  const { mutateAsync: markLockedDistributionDone } =
+    useMarkLockedDistributionDone();
+  return async ({
+    amount,
+    gifts,
+    vault,
+    tokenContractAddress,
+    hedgeyLockPeriod,
+    hedgeyTransferable,
+    epochId,
+    totalGive,
+  }: any) => {
```

Comment:
it will help debugging a lot if this `any` is replaced with proper types

Reply:
I've updated this to have the types